libqtile/bar.py

```diff
@@ -678,5 +682,9 @@ def fake_button_press(self, screen, position, x, y, button=1):
         """
         self.process_button_click(x, y, button)
 
+    def keep_below(self):
+        self.window.keep_below(enable=True)
+        self.window.change_layer(include_internal=True, top_bottom=True, up=False)
```

Comment:
The `change_layer` method here is an X11-specific implementation deal leaking from the backend. Can we avoid that? Perhaps conceptually it would make sense for `self.window.keep_below()` to be responsible for putting it in its place if it isn't already there?